int32 goal_id
int32 goal_state
